mods/boats/init.lua

```diff
@@ -129,25 +132,36 @@ end
 
 function boat.on_step(self, dtime)
 	self.v = get_v(self.object:get_velocity()) * get_sign(self.v)
-	if self.driver then
-		local ctrl = self.driver:get_player_control()
-		local yaw = self.object:get_yaw()
-		if ctrl.up then
-			self.v = self.v + 0.1
+	if self.driver_name then
+		local ctrl = minetest.get_player_by_name(self.driver_name):
```

Comment:
Could return `nil`

Reply:
I'm wondering how? driver name is always a valid player.

Reply:
It's when the player leaves while riding a boat. Bug reproduced using a server & 2nd client which then leaves (does not shut down the server).

Reply:
Yes just realised, sorry. It's also consistent with previous code. Will do.

Reply:
Maybe if the objectref is nil 'driver name' should be set to nil for the now driverless boat?

Reply:
Well, if we're going the 5.0.0-dev way, you could also register `on_detach` callbacks to ensure the driver is set to `nil` when they leave. Your suggestion is good aswell.